docs/mp/server.adoc

```diff
@@ -312,6 +328,74 @@ io.helidon.examples.AdminService:
     path: "/management"
 ----
 
+=== Customizing MP Server at Startup
+
+In the sections above you learned how to configure the WebServer using configuration options and annotations. There are some functions that cannot be handled using the standard configuration options and, instead, must be handled programmatically. 
+
+For example, if you wanted to run code when a CDI container starts, then you would have to write an  _observer method_ in an enabled CDI-managed bean that observes the initialization of the https://jakarta.ee/specifications/cdi/3.0/jakarta-cdi-spec-3.0.html#application_context[application scope]. 
```

Comment:
I feel that we should lead with the example, use a snippet and provide details for the important parts:
- `@Priority(PLATFORM_AFTER)`
- `@Initialized(ApplicationScoped.class`
- access modifier (`private`) 
- Mention `ServerCdiExtension` for advanced use-cases